lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -5088,6 +5088,76 @@ bool QuantizeSwish::run(Function *F, const CompilationContext &cctx) {
   return changed;
 }
 
+/// Fold Exp + ReduceSum + Div into Softmax
+///    IN
+///     |
+///    Exp                IN
+///   /   \                |
+///  |  ReduceSum  -->  Softmax
+///   \   /                |
+///    Div                OUT
+///     |
+///    OUT
+bool OptimizeExpSumDiv::run(Function *F, const CompilationContext &cctx) {
+  LOG_SCOPE(F->getLogContext(), getName());
+
+  bool changed = false;
+  for (auto &N : F->getNodes()) {
+    auto *EN = dyn_cast<ExpNode>(&N);
+    if (!EN || EN->getNumUsers() != 2) {
+      continue;
+    }
+
+    DivNode *DN;
+    BatchedReduceAddNode *RSN;
+    bool optimizationPossible = true;
+
+    auto *user1 = EN->getUsers().front().getUser();
+    auto *user2 = EN->getUsers().back().getUser();
+
+    if (isa<DivNode>(user1) && isa<BatchedReduceAddNode>(user2)) {
+      DN = cast<DivNode>(user1);
+      RSN = cast<BatchedReduceAddNode>(user2);
+    } else if (isa<DivNode>(user2) && isa<BatchedReduceAddNode>(user1)) {
+      DN = cast<DivNode>(user2);
+      RSN = cast<BatchedReduceAddNode>(user1);
+    } else {
+      optimizationPossible = false;
+    }
+
+    if (!optimizationPossible || RSN->getNumUsers() != 1) {
+      continue;
+    }
+
+    // DivNode is a broadcasted node when introduced two extra nodes
+    // for broadcasting and tiling. Ensure that these nodes exist in the
+    // graph.
+    auto *broadcastNode = getOnlyUser(*RSN);
+    if (broadcastNode == nullptr) {
+      continue;
+    }
+    auto *tileNode = getOnlyUser(*broadcastNode);
+    if (tileNode == nullptr) {
+      continue;
+    }
```

Comment:
After you put up this PR an explicit BroadcastNode was added (lowered to Reshape/Tile) in https://github.com/pytorch/glow/pull/5047. Since this pass will occur before lowering, I believe we want to just look for the BroadcastNode instead of the Tile. (If you rebase I imagine your graph opt test will fail)